js/src/modal.js

```diff
@@ -234,6 +245,11 @@ class Modal {
     return config
   }
 
+  // Util worthy?
+  _isTouchDevice() {
+    return 'ontouchstart' in window || window.DocumentTouch && document instanceof window.DocumentTouch
```

Comment:
Can you use the same way we check touch device in our carousel ? 
https://github.com/twbs/bootstrap/blob/v4-dev/js/src/carousel.js#L119

Reply:
done 👍